src/validation/baseValidation.ts

```diff
@@ -0,0 +1,27 @@
+import type { ValidationResult } from '@/contracts/validationResult';
+
+export const noValidation = <T>(_: T): ValidationResult => ({ isValid: true });
+
+export const notNull =
+  (customErrMsg?: string) =>
+  <T>(value: T): ValidationResult => {
+    if (value != null) {
```

Comment:
is this non-strict to also catch `undefined`? Would maybe a simple `!value` do the same? Or do you need `false` and `''` to return `true`?

Reply:
Yes, to catch `undefined` as well, I am not crazy about the `!`. I will change it to a strict check, it makes better sense with the function name 😋

Reply:
Changed to strict 😋